src/the-final-count-down/modules/CountDownClock.js

```diff
@@ -0,0 +1,145 @@
+define([
+    "countdown",
+    "modules/MicroEvent",
+    "loDash"
+], function(countdown, MicroEvent, _){
+
+    /**
+     * A wrapper that providers an API for countdown.js
+     *
+     * @param {Date} date
+     * @param {string} mode
+     * @constructor
+     */
+    var CountDownClock = function(date, mode) {
+
+        // The mode for this instance of the clock
+        this.mode = mode || CountDownClock.MODES.COUNT_DOWN;
+
+        // The date we're counting down to
+        this.date = date;
+
+        // bitwise operator containing the units to be displayed, ie: HOURS, SECONDS, etc
+        this.units = 0;
+
+        // A list of units already added
+        this.includedUnits = [];
+
+        // The interval used by countdown.js
+        this.interval = null;
+    };
+
+    // The counting modes that the clock can run in
+    CountDownClock.MODES = {
+        COUNT_UP:   "COUNT-UP",
+        COUNT_DOWN: "COUNT-DOWN"
+    };
+
+    // Events supported by the clock
+    CountDownClock.EVENTS = {
+        TICK: "COUNTDOWNCLOCK-TICK"
+    };
+
+    CountDownClock.prototype = {
+        /**
+         * Add a unit to be displayed, ie: HOURS, SECONDS, etc
+         *
+         * @param {string} unitConstant
+         */
+        addUnit: function(unitConstant) {
+
+            this.assertClockNotRunning();
+            this.assertValidUnit(unitConstant);
+
+            // If this unit hasn't already been added
+            if (this.includedUnits.indexOf(unitConstant) < 0) {
```

Comment:
I think it would be easier to just have one `units` instance variable that is an array of strings, and then transform it into an OR'd binary value at the least responsible moment, aka when you instantiate the `countdown()`.